internal/gatewayapi/backendtlspolicy.go

```diff
@@ -21,12 +21,79 @@ import (
 	"github.com/envoyproxy/gateway/internal/ir"
 )
 
-func (t *Translator) applyBackendTLSSetting(backendRef gwapiv1.BackendObjectReference, backendNamespace string, parent gwapiv1a2.ParentReference, resources *resource.Resources, envoyProxy *egv1a1.EnvoyProxy) (*ir.TLSUpstreamConfig, error) {
+func (t *Translator) applyBackendTLSSetting(
+	backendRef gwapiv1.BackendObjectReference,
+	backendNamespace string,
+	parent gwapiv1a2.ParentReference,
+	resources *resource.Resources,
+	envoyProxy *egv1a1.EnvoyProxy,
+	isDynamicResolver bool,
+) (*ir.TLSUpstreamConfig, error) {
+	var (
+		err       error
+		tlsBundle *ir.TLSUpstreamConfig
+	)
+
+	// If the destination is a dynamic resolver, we need to use the CACertificateRefs from the backend object
+	// and not from the BackendTLSPolicy. This is because the BackendTLSPolicy requires a valid hostname, and
+	// dynamic resolvers's hostname is not fixed.
+	if isDynamicResolver {
+		if tlsBundle, err = t.processBackendTLSConfig(backendRef, backendNamespace, resources); err != nil {
+			return nil, err
+		}
+		return t.applyEnvoyProxyBackendTLSSetting(tlsBundle, resources, envoyProxy)
+	}
+
 	upstreamConfig, policy, err := t.processBackendTLSPolicy(backendRef, backendNamespace, parent, resources)
 	if err != nil {
 		return nil, err
 	}
-	return t.applyEnvoyProxyBackendTLSSetting(policy, upstreamConfig, resources, parent, envoyProxy)
+	// TODO: zhaohuabing is it correct to surface the EnvoyProxy TLS error to the policy?
+	// We probably should just remove this, the TLS error is already reported in the route status as "InvalidBackendTLS",
+	// and the BackendTLSPolicy is configured correctly.
+	if tlsBundle, err = t.applyEnvoyProxyBackendTLSSetting(upstreamConfig, resources, envoyProxy); err != nil {
+		status.SetTranslationErrorForPolicyAncestors(&policy.Status,
```

Comment:
This keeps the current behavior, however, we probably should just remove this - the TLS error has already been reported in the route status as "InvalidBackendTLS", and this error is not related to the BackendTLSPolicy.